src/coredump/coredump.c

```diff
@@ -706,14 +707,16 @@ static int submit_coredump(
         _cleanup_close_ int coredump_fd = -1, coredump_node_fd = -1;
         _cleanup_free_ char *core_message = NULL, *filename = NULL, *coredump_data = NULL;
         uint64_t coredump_size = UINT64_MAX;
-        bool truncated = false;
+        bool truncated = false, journald_crash;
         int r;
 
         assert(context);
         assert(iovec);
         assert(n_iovec_allocated >= n_iovec + SUBMIT_COREDUMP_FIELDS);
         assert(input_fd >= 0);
 
+        journald_crash = streq_ptr(context[CONTEXT_UNIT], SPECIAL_JOURNALD_SERVICE);
```

Comment:
maybe hide all this in a static mini-function?

```
static bool is_journald_crash(const char *context[_CONTEXT_MAX]) {
        return streq_ptr(context[CONTEXT_UNIT], SPECIAL_JOURNALD_SERVICE);
}
```

And similar for PID 1... Then I think things become a bit more readable, as checks for this would just become:

```c
if (is_journald_crash(context)) {…
```

and the fact how we determine whether something is journald would be hidden. and we could reuse the check later on again

dunno, just an idea...